proxy/src/http/transfer.rs

```diff
@@ -0,0 +1,240 @@
+    transfer_filter(registry, subscriptions)
+}
+
+/// `POST /transfer`
+fn transfer_filter<R: registry::Client>(
+    registry: http::Shared<R>,
+    subscriptions: notification::Subscriptions,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone
+where
+    R: registry::Client,
+{
+    path!("transfer")
+        .and(http::with_shared(registry))
+        .and(http::with_subscriptions(subscriptions))
+        .and(warp::post())
+        .and(path::end())
+        .and(warp::body::json())
+        .and(document::document(document::description("Transfer funds")))
+        .and(document::document(document::tag("Transfer")))
+        .and(document::document(
+            document::body(Input::document()).mime("application/json"),
+        ))
+        .and(document::document(
+            document::response(
+                201,
+                document::body(registry::Transaction::document()).mime("application/json"),
+            )
+            .description("Transfer succeeded"),
+        ))
+        .and_then(handler::transfer)
+}
+
+/// Transfer handlers for conversion between core domain and http request fullfilment.
+mod handler {
+    use std::convert::TryFrom;
+    use warp::http::StatusCode;
+    use warp::{reply, Rejection, Reply};
+
+    use crate::error::Error;
+    use crate::http;
+    use crate::notification;
+    use crate::registry;
+
+    /// Transfer funds to the given `recipient`.
+    pub async fn transfer<R: registry::Client>(
+        registry: http::Shared<R>,
+        subscriptions: notification::Subscriptions,
+        input: super::Input,
+    ) -> Result<impl Reply, Rejection> {
+        // TODO(xla): Get keypair from persistent storage.
+        let fake_pair = radicle_registry_client::ed25519::Pair::from_legacy_string("//Alice", None);
+        // let recipient_key =
+        // radicle_registry_client::ed25519::Public::from_string(input.recipient);
+
+        let reg = registry.write().await;
+        let tx = if let Some(org_id) = input.maybe_org {
+            let org = registry::Id::try_from(org_id).map_err(Error::from)?;
+            reg.transfer_from_org(
+                &fake_pair,
+                org,
+                input.recipient,
+                input.value,
+                input.transaction_fee,
+            )
+            .await?
+        } else {
+            reg.transfer_from_user(
+                &fake_pair,
+                input.recipient,
+                input.value,
+                input.transaction_fee,
+            )
+            .await?
+        };
+
+        subscriptions
+            .broadcast(notification::Notification::Transaction(tx.clone()))
+            .await;
+
+        Ok(reply::with_status(reply::json(&tx), StatusCode::CREATED))
+    }
+}
+
+/// Bundled input data for transfer.
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Input {
+    /// Id if sent from an Org.
+    maybe_org: Option<String>,
```

Comment:
I think because `Id` has a `Deserialize` instance we could use its type here instead of `String`. The `try_from` will be called when trying to deserialize the input then and you don't have to do it above.